lib/index.js

```diff
@@ -315,7 +316,28 @@ Socket.prototype.maxreconn  = opt('maxreconn');
 Socket.prototype.sndprio    = opt('sndprio');
 Socket.prototype.rcvprio    = opt('rcvprio');
 
-/* tcpnodelay sockopt method. this one is a little different */
+/* ipv6 & tcpnodelay sockopt methods. these two opts are a little different */
+Socket.prototype.ipv6 = function (bool) {
+  if(arguments.length){
+    if(bool){
+      if(nn.Setopt(this.binding, nn.NN_SOL_SOCKET, sol.ipv6, 0) > -1)
+        return true;
+      throw new Error(nn.Err() + ': '+this.type + ' ipv6@'+'activing'+'\n');
+    } else {
+      if(nn.Setopt(this.binding, nn.NN_SOL_SOCKET, sol.ipv6, 1) > -1)
+        return false;
+      throw new Error(nn.Err() + ': '+this.type+' ipv6@'+'deactiving'+'\n');
+    }
+  } else {
+    switch(nn.Getopt(this.binding, nn.NN_SOL_SOCKET, sol.ipv6)){
+      case 1: return false;
+      case 0: return true;
+      default:
+        throw new Error(nn.Err() +': '+this.type+' ipv6@'+'getsockopt'+'\n');
```

Comment:
Why for these three errors are we concatenating the last 3 parts of the string?  Why not `... + ipv6@getsockopt\n` ?